src/lib.rs

```diff
@@ -83,6 +101,12 @@ impl Program {
                 &Mul(a, b, c) => locals[a as usize] = (&locals[b as usize] * &locals[c as usize])?,
                 &Div(a, b, c) => locals[a as usize] = (&locals[b as usize] / &locals[c as usize])?,
                 &Rem(a, b, c) => locals[a as usize] = (&locals[b as usize] % &locals[c as usize])?,
+                &Eq(a, b, c) => locals[a as usize] = B(&locals[b as usize] == &locals[c as usize]),
```

Comment:
The derived `PartialEq` impl will consider `I(0)` and `F(0.0)` different. This is probably fine, but it means that we need a numeric cast operator. (Perhaps just make an issue?)